docs: azure-iot-hub: rework docs (#13842)
nrfconnect/sdk-nrf: doc/nrf/libraries/networking/azure_iot_hub.rst

   If the server sends a device-bound message when the device is unavailable for a period of time, for instance while in LTE Power Saving Mode, the server will most likely terminate the TCP connection.
   This will result in additional data traffic as the device has to reconnect to the server, which in turn requires a new TLS handshake and MQTT connection establishment.

.. _prereq_connect_to_azure_iot_hub:

Setup and configuration
***********************

To connect a device to Azure IoT Hub, complete the following steps:

1. :ref:`azure_set_up`
#. :ref:`azure_create_IoT_Hub`
#. :ref:`azure_create_device`
#. :ref:`azure_generating_certificates`
#. :ref:`azure_iot_hub_flash_certs`

.. rst-class:: numbered-step

.. _azure_set_up:

Setting up Azure IoT Hub
========================

If you do not have an Azure account, you need to create one.

To get started with testing the Azure IoT Hub, make sure that the following prerequisites are met:

* Install the `Azure CLI`_.
* To use the ``nrfcredstore`` tool, the dependencies in the :file:`nrf/scripts/requirements-extra.txt` file must be installed.

.. rst-class:: numbered-step

> 

Reviewer: ```suggestion
.. _azure_authenticate:

Authenticate Azure CLI
=================

Authenticate the Azure CLI tool to use your Azure account in the default browser with the following command:

.. code-block:: console

    az login

For other authentication options, see INSERT-LINK-HERE(Sign in with Azure CLI)
https://learn.microsoft.com/en-us/cli/azure/authenticate-azure-cli
```

* For bash scripts, the public device certificate is called :file:`<mydevice>.cert.pem` and is located in a directory called :file:`certs` within the :file:`script` directory.
  The private key is called :file:`<mydevice>.key.pem` and located in a directory called :file:`private` within the :file:`script` directory.

The file and directory names may change if Azure changes their scripts.

.. rst-class:: numbered-step

.. _azure_iot_hub_flash_certs:

Provisioning certificates
=========================

To provision the certificates, use any of the following methods, depending on the DK you are using.

.. tabs::

   .. tab:: nRF91: ``nrfcredstore``

      .. important::
         Program the :ref:`at_client_sample` sample to your device before following this guide and make sure you have ``nrfcredstore`` installed.

      1. Obtain a list of installed keys using the following command:

         .. code-block:: console

            nrfcredstore <serial port> list

         where ``<serial port>`` is the serial port of your device.

      #. Provision the private key to the modem and replace the placeholders:

         .. code-block:: console

>             nrfcredstore <serial port> write <sec tag> CLIENT_KEY {device_name}.key

Reviewer: ```suggestion
            nrfcredstore <serial port> write <sec tag> CLIENT_KEY {device_name}.key.pem
```

      #. Provision the client certificate and replace the placeholders:

         .. code-block:: console

>             nrfcredstore <serial port> write <sec tag> CLIENT_CERT {device_name}.key

Reviewer: ```suggestion
            nrfcredstore <serial port> write <sec tag> CLIENT_CERT {device_name}.cert.pem
```

>       #. Provison the server certificate, which you downloaded previously, by running one of the following commands:
> 
>          .. code-block:: console
> 
>             nrfcredstore <serial port> write <sec tag> ROOT_CA_CERT DigiCertGlobalRootG2.crt.pem
> 
>          .. code-block:: console
> 
>             nrfcredstore <serial port> write <sec tag> ROOT_CA_CERT BaltimoreCyberTrustRoot.crt.pem

Reviewer: ```suggestion
      #. Provison the server certificates, which you downloaded previously, by running the following commands:

         .. code-block:: console

            nrfcredstore <serial port> write <sec tag> ROOT_CA_CERT DigiCertGlobalRootG2.crt.pem

         .. code-block:: console

            nrfcredstore <serial port> write <secondary sec tag> ROOT_CA_CERT BaltimoreCyberTrustRoot.crt.pem
```

To connect to Azure IoT Hub without using DPS, complete the following minimum required configuration:

> 1. In the `Azure Portal`_, navigate to :guilabel:`IoT Hub` and select the desired IoT Hub.

Reviewer: ```suggestion
1. To retrieve your IoT Hub hostname, run the following command:
    .. code-block:: console

        az iot hub show --name <hub_name> --query "properties.hostName"

#. Configure :kconfig:option:`CONFIG_AZURE_IOT_HUB_HOSTNAME` to the returned address.
```

Wasn't able to remove the next line because it was not changed in the PR, but it should be removed after this change.

Author: Thanks! Applied this and the other suggestions, and removed the line below
